components/clp-package-utils/clp_package_utils/scripts/archive_manager.py

```diff
@@ -0,0 +1,234 @@
+    )
+    find_parser: argparse.ArgumentParser = subparsers.add_parser(
+        FIND_COMMAND,
+        help="Lists IDs of compressed archives.",
+    )
+    del_parser: argparse.ArgumentParser = subparsers.add_parser(
+        DEL_COMMAND,
+        help="Deletes compressed archives from the database and file system.",
+    )
+
+    # Options for find subcommand
+    find_parser.add_argument(
+        BEGIN_TS_ARG,
+        dest="begin_ts",
+        type=int,
+        default=0,
+        help="Time-range lower-bound (inclusive) as milliseconds from the UNIX epoch.",
+    )
+    find_parser.add_argument(
+        END_TS_ARG,
+        dest="end_ts",
+        type=int,
+        help="Time-range upper-bound (include) as milliseconds from the UNIX epoch.",
+    )
+
+    # Options for delete subcommand
+    del_parser.add_argument(
+        DRY_RUN_ARG,
+        dest="dry_run",
+        action="store_true",
+        help="Only prints the archives to be deleted, without actually deleting them.",
+    )
+
+    # Subcommands for delete subcommand
+    del_subparsers: argparse._SubParsersAction[argparse.ArgumentParser] = del_parser.add_subparsers(
+        dest="del_subcommand",
+        required=True,
+    )
+
+    # Delete by ID subcommand
+    del_id_parser: argparse.ArgumentParser = del_subparsers.add_parser(
+        DEL_BY_IDS_SUBCOMMAND,
+        help="Deletes archives by ID.",
+    )
+
+    # Delete by ID arguments
+    del_id_parser.add_argument(
+        "ids",
+        nargs="+",
+        help="List of archive IDs to delete",
+    )
+
+    # Delete by filter subcommand
+    del_filter_parser: argparse.ArgumentParser = del_subparsers.add_parser(
+        DEL_BY_FILTER_SUBCOMMAND,
+        help="Deletes compressed archives that fall within the specified time range.",
+    )
+
+    # Delete by filter arguments
+    del_filter_parser.add_argument(
+        BEGIN_TS_ARG,
+        type=int,
+        default=0,
+        help="Time-range lower-bound (inclusive) as milliseconds from the UNIX epoch.",
+    )
+    del_filter_parser.add_argument(
+        END_TS_ARG,
+        type=int,
+        required=True,
+        help="Time-range upper-bound (include) as milliseconds from the UNIX epoch.",
+    )
+
+    parsed_args: argparse.Namespace = args_parser.parse_args(argv[1:])
+
+    begin_timestamp: int
+    end_timestamp: int
+    subcommand: str = parsed_args.subcommand
+
+    # Validate and load config file
+    try:
+        config_file_path: Path = Path(parsed_args.config)
+        clp_config: CLPConfig = load_config_file(
+            config_file_path, default_config_file_path, clp_home
+        )
+        clp_config.validate_logs_dir()
+
+        # Validate and load necessary credentials
+        validate_and_load_db_credentials_file(clp_config, clp_home, False)
+    except:
+        logger.exception("Failed to load config.")
+        return -1
+
+    storage_type: StorageType = clp_config.archive_output.storage.type  # Unsure about type
```

Comment:
Huh, I feel the Literal has introduces some complication.. 
I would say just use StorageType for now. Planning to get rid of that Literal if possible.